drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -507,15 +507,30 @@ def external_form_object_key=(value)
     super(value.blank? ? nil : value.strip)
   end
 
-  def walk_definition_nodes(&block)
-    definition_struct&.item&.each do |node|
-      walk_definition_node(node, &block)
+  # Walk definition items either as open structs or hashes
+  def walk_definition_nodes(as_open_struct: false, &block)
+    if as_open_struct
```

Comment:
This is fine, I guess I'm not sold that the utility of the struct variant is worth the additional complexity but we can leave it as is and refactor later